### DISCUSSION TITLE
Repository source link for the dependency file.

### DISCUSSION THREAD

**Discussion by ashfaq-shamsudeen** [DependencyTrack/dependency-track#1094] [Q&A]
Is there a way to get the repository source link for the dependency file e.g. bytebuddy dependency and the repository link would be https://mvnrepository.com/artifact/net.bytebuddy/byte-buddy? Thanks for considering!

**Comment by stevespringett** [comment]
If the component has a Package URL, then yes, you can determine where the package location is. Per the purl spec, Maven Central is the default repo for Maven components. So any component without an alternative `repository_url` would be using Central. For the components that have a `repository_url` specified, you should use that value.

Also, please note that https://mvnrepository.com/artifact/net.bytebuddy/byte-buddy isn't a repo link. It's not even a repo. Its an index to other repos. In this example, bytebuddy is found in four different public repos including Central, Redhat GA, Redhat EA, and ICM.

An example purl for bytebuddy that is pulled from Redhat GA repo for example, would be:
```
pkg:maven/net.bytebuddy/byte-buddy@1.10.17.redhat-00004?repository_url=maven.repository.redhat.com/ga/
```

**Comment by ashfaq-shamsudeen** [comment]
Thank you so much for your valuable comments and this guides me in the
right direction! Sorry for the late reply as I was down and out.
